### PR TITLE
Integrate vic-ui into UI installer script

### LINKED ISSUE #1101: vic-machine install-ui
This describes the deployment process for installing the UI plugin via `vic-machine`

### Overview
* User downloads VIC tarball and successfully runs `vic-machine create`
    * Tarball includes VIC UI plugin server container and VIC UI plugin zip
    * Plugin server container is added to imagec
* User runs `vic-machine install-ui`
    * Run a plugin server container that serves the VIC UI plugin zip over HTTP
    * Obtain the IP of the container
    * Generate the plugin manifest with the URL of the VIC UI plugin on the plugin server container
    * Upload the manifest to vSphere (registers the plugin)
    * Trigger the plugin install by logging into the UI

### Requirements
* Must be able to install the UI plugin without internet access (many customers do not have internet access in their datacenter)

### Known work
* Translate plugin registration (jar exists) to govmomi commands

### Research
* How to distribute a container image in the VIC tarball
* How to put container image directly into imagec
* How to get vic-machine to login to the UI to trigger plugin install

### Example command
```vic-machine install-ui --target <vSphere IP> --user <username> --password <password> --vch-ip <VCH IP>```

### PR DESCRIPTION
**Pull request by jooskim**
This is a two-fold PR in that

1) The installer script now uses vic-ui binary for registering/unregistering the VIC UI plugin and this removes the JRE requirement. Fixes #1101

2) This is a workaround for the issue found in #1668. Installer script detects if the target VCSA has the bash shell accessible through a direct SSH command and will display detailed instructions on how to manually complete the installation procedure if shell is not available. @stuclem Please review the commit message of https://github.com/jooskim/vic/commit/fb0b89b95854fdeee5fdc60a0b4ff33d06cad928.

### CODE REVIEW

File: ui/installer/VCSA/install.sh [hunk 1]
```diff
@@ -131,30 +153,66 @@ upload_packages () {
...
+        scp -qr $PLUGIN_BUNDLES root@$VCENTER_IP:/tmp/
+        if [[ $? > 0 ]] ; then
+            echo "Error! Could not upload the VIC plugins to the target VCSA"
+            exit 1
+        fi
+    else
+        echo "-------------------------------------------------------------"
+        echo "WARNING!! Bash shell is required to complete the installation"
+        echo "automatically, however it is not enabled on this machine."
+        echo "Installation cannot be completed manually. Please follow the"
```
Comment (andrewtchin) [inline ui/installer/VCSA/install.sh]: Should this be `can be completed manually` or `cannot be completed automatically`?

Comment (jooskim) [inline ui/installer/VCSA/install.sh]: Oops good catch. That's a typo. I'll fix that

**Comment by andrewtchin** [comment]
lgtm

File: ui/installer/VCSA/install.sh [hunk 2]
```diff
@@ -170,6 +228,9 @@ if [[ $VIC_UI_HOST_URL == "NOURL" ]] ; then
     update_ownership
 fi
 
+cleanup
+rm extension_already_registered
+
 echo "--------------------------------------------------------------"
 if [[ $? > 0 ]] ; then
```
Comment (mdubya66) [inline ui/installer/VCSA/install.sh]: $? will always be 0 because the echo will always succeed.

Comment (jooskim) [inline ui/installer/VCSA/install.sh]: Right, that's a silly mistake. I'll fix that.

**Comment by emlin** [comment]
lgtm 👍

**Comment by jooskim** [comment]
♻️

**Pull request merged**